java/src/main/java/ai/rapids/cudf/WindowOptions.java

```diff
@@ -158,21 +164,48 @@ public Builder window(ColumnVector precedingCol, ColumnVector followingCol) {
       return this;
     }
 
+    /**
+     * @deprecated use orderByColumnIndex(int index)
+     * @return
+     */
+    @Deprecated
```

Comment:
I am grateful that you're doing this via deprecation, and that you are routing both timestamp and non-timestamp calls through the same API. 
This might be the best approach. It will allow us to remove the old code from the C++ side more aggressively.